packages/design-toolkit/src/components/flashcard/index.tsx

```diff
@@ -0,0 +1,200 @@
+import 'client-only';
+import { createContext, useContext } from 'react';
+import { Skeleton } from '../skeleton';
+import { FlashcardStyles } from './styles';
+import type {
+  FlashcardComponentProps,
+  FlashcardDetailContainerProps,
+  FlashcardProps,
+} from './types';
+
+const {
+  container,
+  hero,
+  identifier,
+  header,
+  subHeader,
+  secondaryContainer,
+  secondaryData,
+  detailsContainer,
+  detailsItem,
+  detailsLabel,
+  detailsValue,
+  skeleton,
+} = FlashcardStyles();
+
+export const FlashcardContext = createContext<FlashcardProps>({
+  isLoading: false,
+});
+
+export function Flashcard(props: FlashcardProps) {
+  const { isLoading, children, className, ...rest } = props;
+
+  return (
+    <FlashcardContext.Provider value={{ isLoading }}>
+      <div {...rest} className={container({ className })}>
+        {children}
+      </div>
+    </FlashcardContext.Provider>
+  );
+}
+Flashcard.displayName = 'Flashcard';
+
+export function FlashcardHero(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  if (isLoading) {
+    return (
+      <div {...rest} className={hero({ className })}>
+        <Skeleton className={skeleton()} />
+        <Skeleton className={skeleton({ className: 'max-w-1/2' })} />
+      </div>
+    );
+  }
+
+  return (
+    <div {...rest} className={hero()}>
+      {children}
+    </div>
+  );
+}
+FlashcardHero.displayName = 'FlashcardHero';
+
+export function FlashcardIdentifier(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={identifier({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardIdentifier.displayName = 'FlashcardIdentifier';
+
+export function FlashcardHeader(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={header({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardHeader.displayName = 'FlashcardHeader';
+
+export function FlashcardSubheader(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={subHeader({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardSubheader.displayName = 'FlashcardSubheader';
+
+export function FlashcardSecondary(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={secondaryContainer({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardSecondary.displayName = 'FlashcardSecondary';
+
+export function FlashcardSecondaryData(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  // While loading, don't display.
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <h2 {...rest} className={secondaryData({ className })}>
```

Comment:
doesn't make sense to have an h2 here when we're formatting the text style. swtich to a `<p>` or `<div>` tag